src/history/hs.c

```diff
@@ -666,6 +670,32 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
             if (upd->txnid == WT_TXN_ABORTED)
                 continue;
             WT_ERR(__wt_modify_vector_push(&modifies, upd));
+
+            /*
+             * If the update is the second update older than the prepared update and we haven't seen
+             * a tombstone. Mark the update.
+             */
+            if (upd->prepare_state == WT_PREPARE_INPROGRESS) {
+                /*
+                 * No normal update between prepared updates and the first prepared update cannot be
+                 * a tombstone.
+                 */
+                WT_ASSERT(session, (track_prepare && upd_count == 0) ||
+                    (!track_prepare && upd->type != WT_UPDATE_TOMBSTONE));
+                track_prepare = true;
+            } else if (track_prepare) {
+                if (upd->type == WT_UPDATE_TOMBSTONE) {
+                    upd_count = 0;
+                    track_prepare = false;
+                } else if (upd_count == 0)
+                    ++upd_count;
+                else {
+                    second_older_than_prepare = upd;
+                    upd_count = 0;
```

Comment:
Do we need to reset the counter?

Reply:
I'm interested in this too. What if we hit this condition again?